src/java/picard/vcf/PairedVariantSubContextIterator.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * */
+public class PairedVariantSubContextIterator implements Iterator<PairedVariantSubContextIterator.VcfTuple> {
+    private final PeekableIterator<VariantContext> leftIterator;
+    private final String leftSample;
+    private final PeekableIterator<VariantContext> rightIterator;
+    private final String rightSample;
+    private final VariantContextComparator comparator;
+
+    public PairedVariantSubContextIterator(final Iterator<VariantContext> leftIterator, final String leftSample,
+                                    final Iterator<VariantContext> rightIterator, final String rightSample,
+                                    final SAMSequenceDictionary dict) {
+        this.leftIterator  = new PeekableIterator<>(leftIterator);
+        this.leftSample    = leftSample;
+        this.rightIterator = new PeekableIterator<>(rightIterator);
+        this.rightSample   = rightSample;
+        this.comparator    = new VariantContextComparator(dict);
+    }
+
+    @Override
+    public boolean hasNext() {
+        return this.leftIterator.hasNext() || this.rightIterator.hasNext();
+    }
+
+    @Override
+    public VcfTuple next() {
+        if (!hasNext()) throw new IllegalStateException("next() called while hasNext() is false.");
+
+        final Optional<VariantContext> leftVariantContext  = this.leftIterator.hasNext() ? Optional.of(this.leftIterator.peek()) : Optional.empty();
+        final Optional<VariantContext> rightVariantContext = this.rightIterator.hasNext() ? Optional.of(this.rightIterator.peek()) : Optional.empty();
+
+        // If one or the other is missing because there is no next, just return a one-sided tuple
+        if (!leftVariantContext.isPresent() && !rightVariantContext.isPresent()) {
+            throw new IllegalStateException("BUG: Both contexts empty.");
+        }
+        else if (!leftVariantContext.isPresent()) {
+            return new VcfTuple(Optional.empty(), this.rightIterator.next().subContextFromSample(rightSample));
+        }
+        else if (!rightVariantContext.isPresent()) {
+            return new VcfTuple(this.leftIterator.next().subContextFromSample(leftSample), Optional.empty());
+        }
+        else { // Otherwise check the ordering and do the right thing
+            final int ordering = this.comparator.compare(leftVariantContext.get(), rightVariantContext.get());
+            if (ordering == 0) {
+                return new VcfTuple(this.leftIterator.next().subContextFromSample(leftSample), this.rightIterator.next().subContextFromSample(rightSample));
+            } else if (ordering < 0) {
+                return new VcfTuple(this.leftIterator.next().subContextFromSample(leftSample), Optional.empty());
+            } else {
+                return new VcfTuple(Optional.empty(), this.rightIterator.next().subContextFromSample(rightSample));
+            }
+        }
+    }
+
+    @Override public void remove() {
```

Comment:
Should the override annotation be on a separate line (not sure of the convention in this codebase)?